dashboard/app/models/circuit_playground_discount_code.rb

```diff
@@ -20,11 +20,21 @@ def self.claim(full_discount)
 
     code = nil
     Retryable.retryable(on: ActiveRecord::RecordNotSaved) do
-      code = where(full_discount: full_discount).
+      codes = where(full_discount: full_discount).
         where(claimed_at: nil).
         where(voided_at: nil).
         where(expiration_field.gt(Time.now)).
-        first
+        limit(20)
+
+      if codes.length < 20
+        Honeybadger.notify("Fewer than 20 remaining circuit playground discount codes",
```

Comment:
This call looks right.

Reply:
Actually, without an error (exception) object, the call is slightly different in order to get it to show up the way we expect. This will work, but I don't think the message will show up in the expected place. @drewsamnick and I went through this recently and came up with this https://github.com/code-dot-org/code-dot-org/blob/e9a1b2cec851f49fd39e69332f4441bac00c6f91/dashboard/app/models/school_info.rb#L116-L122

The main difference is you want to add an `error_class`, which here is just an arbitrary string, and explicitly specify that the supplied message here is an `error_message`.

If you want, you can test this by setting up a personal honeybadger project and logging there from development. I did that a couple years ago when I set up the cronjob-honeybadger logging (aka cronbadger).

Reply:
I realize I sent you examples that supplied a rescued error, so those were slightly misleading in this context.